siamese_network/README.md

```diff
@@ -0,0 +1,7 @@
+# Siamese Network Example
```

Comment:
OK last thing can you please quickly explain what a Siamese network is here, the high-level architecture and what users should expect to input and get as an output for this example. Not a tutorial.

Also, post some proof in a screenshot here on the PR that this thing works

Reply:
I have included an explanation in the class definition. I will update the read me to include the same definition.

Here is a screenshot to prove that the model works using the default setup.

![Screenshot from 2022-05-10 21-44-58](https://user-images.githubusercontent.com/17136125/167770941-47cadc28-2a8a-438a-bd40-eb551cd6d3af.png)